sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -771,8 +771,8 @@ object SymmetricHashJoinStateManager {
       joinSide: JoinSide,
       inputValueAttributes: Seq[Attribute],
       joinKeys: Seq[Expression],
-      stateFormatVersion: Int): scala.collection.mutable.Map[String, (StructType, StructType)] = {
-    val result = scala.collection.mutable.Map[String, (StructType, StructType)]()
+      stateFormatVersion: Int): Map[String, (StructType, StructType)] = {
+    var result: Map[String, (StructType, StructType)] = Map.empty
```

Comment:
I meant about using mutable Map to calculate without copying but at the end converting to immutable Map. But I don't think this is a part of critical path, so OK.

Reply:
yea cool - will keep this as it is then